src/udev/udev-builtin-net_id.c

```diff
@@ -163,8 +163,29 @@ static bool is_valid_onboard_index(unsigned long idx) {
         return idx <= (naming_scheme_has(NAMING_16BIT_INDEX) ? ONBOARD_16BIT_INDEX_MAX : ONBOARD_14BIT_INDEX_MAX);
 }
 
+static int parse_vf_repr_num(const LinkInfo *info, unsigned *num) {
+        unsigned pf;
+
+        assert(info);
+
+        if (!isempty(info->phys_port_name) &&
+            sscanf(info->phys_port_name, "pf%uvf%u", &pf, num) == 2)
+                return 0;
+
+        return -EINVAL;
+}
+
+static bool is_vf_repr(const LinkInfo *info) {
+        unsigned vf;
+
+        assert(info);
+
+        return !parse_vf_repr_num(info, &vf);
```

Comment:
Our coding style says that `!` should only be used with boolean or boolean-like values. Please use '== 0' here.

EDIT: or just drop the function, see other comments.